raft/src/main/java/org/apache/kafka/raft/RaftConfig.java

```diff
@@ -34,50 +39,63 @@
+
+
+    public static final String QUORUM_VOTERS_CONFIG = QUORUM_PREFIX + "bootstrap.voters";
     private static final String QUORUM_VOTERS_DOC = "List of voters. This is only used the " +
         "first time a cluster is initialized.";
 
-    public static final String QUORUM_ELECTION_TIMEOUT_MS_CONFIG = "quorum.election.timeout.ms";
+    public static final String QUORUM_ELECTION_TIMEOUT_MS_CONFIG = QUORUM_PREFIX + "election.timeout.ms";
     private static final String QUORUM_ELECTION_TIMEOUT_MS_DOC = "Maximum time in milliseconds to wait " +
         "without being able to fetch from the leader before triggering a new election";
 
-    public static final String QUORUM_ELECTION_JITTER_MAX_MS_CONFIG = "quorum.election.jitter.max.ms";
+    public static final String QUORUM_ELECTION_JITTER_MAX_MS_CONFIG = QUORUM_PREFIX + "election.jitter.max.ms";
     private static final String QUORUM_ELECTION_JITTER_MAX_MS_DOC = "Maximum jitter to delay new elections. " +
         "This helps prevent gridlocked elections";
 
+    public static final String QUORUM_FETCH_TIMEOUT_MS_CONFIG = QUORUM_PREFIX + "fetch.timeout.ms";
+    private static final String QUORUM_FETCH_TIMEOUT_MS_DOC = "Maximum time without a successful fetch from " +
+        "the current leader before a new election is started";
+
+
     static {
         CONFIG = new ConfigDef()
-            .define(BOOTSTRAP_SERVERS_CONFIG,
+            .define(QUORUM_PREFIX + BOOTSTRAP_SERVERS_CONFIG,
                 ConfigDef.Type.LIST,
                 Collections.emptyList(),
                 new ConfigDef.NonNullValidator(),
                 ConfigDef.Importance.HIGH,
                 CommonClientConfigs.BOOTSTRAP_SERVERS_DOC)
-            .define(REQUEST_TIMEOUT_MS_CONFIG,
+            .define(QUORUM_PREFIX + REQUEST_TIMEOUT_MS_CONFIG,
                 ConfigDef.Type.INT,
                 20000,
                 atLeast(0),
                 ConfigDef.Importance.MEDIUM,
                 REQUEST_TIMEOUT_MS_DOC)
-            .define(RETRY_BACKOFF_MS_CONFIG,
-                ConfigDef.Type.LONG,
-                100L,
+            .define(QUORUM_PREFIX + RETRY_BACKOFF_MS_CONFIG,
+                ConfigDef.Type.INT,
+                100,
                 atLeast(0L),
                 ConfigDef.Importance.LOW,
                 CommonClientConfigs.RETRY_BACKOFF_MS_DOC)
             .define(QUORUM_VOTERS_CONFIG,
                 ConfigDef.Type.LIST,
                 null,
-                new ConfigDef.NonEmptyString(),
                 ConfigDef.Importance.HIGH,
                 QUORUM_VOTERS_DOC)
             .define(QUORUM_ELECTION_TIMEOUT_MS_CONFIG,
                 ConfigDef.Type.INT,
-                30000,
+                5000,
                 atLeast(0L),
                 ConfigDef.Importance.HIGH,
                 QUORUM_ELECTION_TIMEOUT_MS_DOC)
+            .define(QUORUM_FETCH_TIMEOUT_MS_CONFIG,
+                ConfigDef.Type.INT,
+                15000,
```

Comment:
From the paper (https://www.cl.cam.ac.uk/~ms705/pub/papers/2015-osr-raft.pdf) the fetch timeout is suggested with 150ms while election timeout is suggested with current broadcastTime which could be smaller than 150 in normal networks. Personally I also feel that the default timeout for election timeout should be in the range of smaller than 100ms, while fetch timeout should be within several seconds.

Reply:
It may take some experimentation to get these defaults right. In the unstable environments that we run Kafka in these days, I have a hard time seeing anything less than 10s being stable. I think the fetch timeout should probably be on the same order as the zk session timeout, which is now 18s by default. However, I agree that the election timeout can probably go a bit lower.

Reply:
Sounds good.